cpp/src/slice2py/Python.cpp

```diff
@@ -42,388 +43,67 @@ namespace
-    // its submodules.
-    //
-    // The PackageVisitor class creates the directory hierarchy to mirror the
-    // Slice module hierarchy, and updates the __init__.py files as necessary.
-    //
-    class PackageVisitor final : public ParserVisitor
-    {
-    public:
-        static void createModules(const UnitPtr&, const string&, const string&);
-
-        void visitModuleEnd(const ModulePtr&) final;
-
-    private:
-        PackageVisitor(StringList&);
-
-        enum ReadState
-        {
-            PreModules,
-            InModules,
-            InSubmodules
-        };
-
-        static void addModule(const string&, const string&, const string&);
-        static void addSubmodule(const string&, const string&, const string&);
-
-        static void readInit(const string&, StringList&, StringList&);
-        static void writeInit(const string&, const string&, const StringList&, const StringList&);
-
-        StringList& _modules;
-    };
-
-    const string moduleTag = "# Modules:";       // NOLINT(cert-err58-cpp)
-    const string submoduleTag = "# Submodules:"; // NOLINT(cert-err58-cpp)
-
-    PackageVisitor::PackageVisitor(StringList& modules) : _modules(modules) {}
-
-    void PackageVisitor::createModules(const UnitPtr& unt, const string& module, const string& dir)
-    {
-        StringList modules;
-        PackageVisitor v(modules);
-        unt->visit(&v);
-
-        for (const auto& p : modules)
-        {
-            vector<string> vs;
-            if (!IceInternal::splitString(p, ".", vs))
-            {
-                assert(false);
-            }
-            string currentModule;
-            string path = dir.empty() ? "." : dir;
-            for (auto q = vs.begin(); q != vs.end(); ++q)
-            {
-                if (q != vs.begin())
-                {
-                    addSubmodule(path, currentModule, *q);
-                    currentModule += ".";
-                }
-
-                currentModule += *q;
-                path += "/" + *q;
-                createDirectory(path);
-
-                addModule(path, currentModule, module);
-            }
-        }
-    }
-
-    void PackageVisitor::visitModuleEnd(const ModulePtr& p)
-    {
-        //
-        // Collect the most deeply-nested modules. For example, if we have a
-        // module named M.N.O, then we don't need to keep M or M.N in the list.
-        //
-        string abs = getAbsolute(p);
-        if (find(_modules.begin(), _modules.end(), abs) == _modules.end())
-        {
-            _modules.push_back(abs);
-        }
-        string::size_type pos = abs.rfind('.');
-        if (pos != string::npos)
-        {
-            string parent = abs.substr(0, pos);
-            _modules.remove(parent);
-        }
-    }
+        // Create a new output file for this package.
+        string fileName = packageName;
+        replace(fileName.begin(), fileName.end(), '.', '/');
+        fileName += "/__init__.py";
```

Comment:
Should we be using `IceInternal::separator`? Or does `/` work fine?

Reply:
Seems to work fine, that being say using `IceInternal::separator` seems the right thing to do. I will update the code.